core/controllers/base.py

```diff
@@ -215,6 +220,15 @@ def dispatch(self):
                 b'https://oppiatestserver.appspot.com', permanent=True)
             return
 
+        if feconf.ENABLE_MAINTENANCE_MODE and not self.is_super_admin:
```

Comment:
Just a question about this. Suppose the user is a super admin but is not logged in (or gets logged out somehow). They'll need to log in somehow. Will they be able to do so?

(Perhaps sufficient for them to just go to /admin, but I just wanted to check that that would still work even if the user isn't logged in already. Probably should have a backend integration test for this case.)